packages/network-controller/tests/helpers.ts

```diff
@@ -0,0 +1,145 @@
+import type { InfuraNetworkType } from '@metamask/controller-utils';
+import {
+  BUILT_IN_NETWORKS,
+  isInfuraNetworkType,
+  toHex,
+} from '@metamask/controller-utils';
+
+import { FakeBlockTracker } from '../../../tests/fake-block-tracker';
+import { FakeProvider } from '../../../tests/fake-provider';
+import type { FakeProviderStub } from '../../../tests/fake-provider';
+import type {
+  BuiltInNetworkClientId,
+  CustomNetworkClientId,
+  NetworkClient,
+  NetworkClientConfiguration,
+  NetworkClientId,
+  NetworkController,
+} from '../src';
+import type { AutoManagedNetworkClient } from '../src/create-auto-managed-network-client';
+import type {
+  CustomNetworkClientConfiguration,
+  InfuraNetworkClientConfiguration,
+} from '../src/types';
+import { NetworkClientType } from '../src/types';
+
+/**
+ * Builds an object that satisfies the NetworkClient shape, but using a fake
+ * provider and block tracker which doesn't make any requests.
+ *
+ * @param args - Arguments to this function.
+ * @param args.configuration - The desired network client configuration.
+ * @param args.providerStubs - Objects that allow for stubbing specific provider
+ * requests.
+ * @returns The fake network client.
+ */
+function buildFakeNetworkClient({
```

Comment:
This pattern of including test helpers specific to a controller inside the package for that controller is new. Previously we might have just put it within the test file that needed these helpers. But by putting them here, we define them once and then any package in the monorepo can use them.